File: C/Linear_Search.c
#include<stdio.h>

int main(){
    int i, n, x, found=0; //declaration
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    int ar[n];
    printf("Enter %d numbers\n", n);
    for(i=0; i<n; i++)  //input elements
        scanf("%d", &ar[i]);
    printf("Enter element to be searched for: ");
    scanf("%d", &x);    
    for(i=0; i<n; i++) //searching for x
        if(ar[i]==x){   //if found
            found=1;
            printf("Element found at position %d", i+1);
            break;
        }
    if(found==0) //if not found
        printf("Element not found");        
}

/*
    Problem name :- Linear Search
    Example 1:
    Input:-
    Enter the number of elements: 5
    Enter 5 numbers
    10
    15
    35
    74
    8
    56
    Enter element to be searched for: 74

    Output:-
    Element found at position 4


    Example 2:
    Input:-
    Enter the number of elements: 5
    Enter 5 numbers
    10
    15
    35
    74
    8
    56
    Enter element to be searched for: 75
    
    Output:-
    Element not found

    
    Time Complexity:- O(n)
    Space Complexity:- O(1)
*/

File: JavaScript/linearSearch.js
//Linear Search

function linear_search(arr,k){
    for(i=0;i<arr.length;i++){
        if(k==arr[i])
        {
            return i+1;
        }
    }
    return -1;
}

//Example
console.log(linear_search([3,4,5,6,7,2,9,0],6)); // 4

/* Linear Search
Input : Array and key to be searched 
Output : Returns position of first key present in array
          Returns -1 if key not present in array
Time Complexity: O(n)
Space Complexity: O(1)
*/